build/Build.cs

```diff
@@ -0,0 +1,137 @@
+
+    Target Clean => _ => _
+        .Before(Restore)
+        .Executes(() =>
+        {
+            SourceDirectory.GlobDirectories("**/bin", "**/obj", "./source/**/TestResults").ForEach(DeleteDirectory);
+            EnsureCleanDirectory(ArtifactsDirectory);
+        });
+
+    Target Restore => _ => _
+        .DependsOn(Clean)
+        .Executes(() =>
+        {
+            DotNetRestore(s => s
+                .SetProjectFile(Solution));
+        });
+
+    Target Compile => _ => _
+        .DependsOn(Restore)
+        .Executes(() =>
+        {
+            DotNetBuild(s => s
+                .SetProjectFile(Solution)
+                .SetConfiguration(Configuration)
+                .SetVersion(OctoVersionInfo.FullSemVer)
+                .SetAssemblyVersion(OctoVersionInfo.MajorMinorPatch)
+                .SetInformationalVersion(OctoVersionInfo.InformationalVersion)
+                .EnableNoRestore());
+        });
+
+    Target Test => _ => _
+        .DependsOn(Compile)
+        .Executes(() =>
+        {
+            DotNetTest(_ => _
+                .SetProjectFile(Solution)
+                .SetConfiguration(Configuration)
+                .SetLoggers("trx")
+                .SetVerbosity(DotNetVerbosity.Normal)
+                .SetFilter(TestFilter)
+                .EnableNoBuild()
+                .EnableNoRestore());
+            GlobFiles(SourceDirectory, "**/*.trx")
+                .ForEach(x => CopyFileToDirectory(x, ArtifactsDirectory, FileExistsPolicy.Overwrite));
+        });
+
+    Target Pack => _ => _
+        .DependsOn(Compile)
+        .DependsOn(Test)
+        .Executes(() =>
+        {
+            DotNetPack(_ => _
+                .SetProject(Solution)
+                .SetConfiguration(Configuration)
+                .SetOutputDirectory(ArtifactsDirectory)
+                .EnableNoBuild()
+                .AddProperty("Version", OctoVersionInfo.FullSemVer)
+            );
+        });
+
+    Target Publish => _ => _
+        .OnlyWhenStatic(() => IsServerBuild)
+        .DependsOn(Pack)
+        .TriggeredBy(Pack)
+        .Executes(() =>
+        {
+            var packageName = Solution.Octodiff.Name;
+            NuGetTasks.NuGetPush(_ => _
+                .SetTargetPath(ArtifactsDirectory / $"{packageName}.{OctoVersionInfo.FullSemVer}.nupkg")
+                .SetSource("https://f.feedz.io/octopus-deploy/dependencies/nuget")
+                .SetApiKey(FeedzIoApiKey)
+            );
+            if (string.IsNullOrWhiteSpace(OctoVersionInfo.PreReleaseTagWithDash))
+            {
+                NuGetTasks.NuGetPush(_ => _
+                    .SetTargetPath(ArtifactsDirectory / $"{packageName}.{OctoVersionInfo.FullSemVer}.nupkg")
+                    .SetSource("https://www.nuget.org/api/v2/package")
+                    .SetApiKey(NuGetApiKey));
+            }
+        });
+
+    Target CopyToLocalPackages => _ => _
+        .OnlyWhenStatic(() => IsLocalBuild)
+        .TriggeredBy(Compile)
```

Comment:
Same as above triggeredBy

Reply:
```suggestion
        .TriggeredBy(Pack)
```

Reply:
I've changed this from `Compile` to `Pack`, but I'm not sure it's addressed your question